build_tools/benchmarks/collect_compilation_statistics.py

```diff
@@ -130,97 +133,169 @@ def get_module_path(flag_file: TextIO) -> Optional[str]:
+        model_tags=tuple(model.tags),
+        model_source=model.source_type.value,
+        target_arch=f"[{','.join(target_archs)}]",
+        compile_tags=tuple(compile_config.tags))
+    module_dir_path = iree_artifacts.get_module_dir_path(
+        module_generation_config=gen_config, root_path=e2e_test_artifacts_dir)
+    module_path = module_dir_path / iree_artifacts.MODULE_FILENAME
+    module_map[compilation_info] = pathlib.Path(module_path)
+
+  return module_map
+
+
+def get_module_map_from_benchmark_suite(
+    benchmark_suite_dir: pathlib.Path) -> Dict[CompilationInfo, pathlib.Path]:
+  benchmark_suite = BenchmarkSuite.load_from_benchmark_suite_dir(
+      benchmark_suite_dir)
+  module_map = {}
+  for category, _ in benchmark_suite.list_categories():
+    benchmark_cases = benchmark_suite.filter_benchmarks_for_category(
+        category=category)
+    for benchmark_case in benchmark_cases:
+      if benchmark_case.benchmark_case_dir is None:
+        raise ValueError("benchmark_case_dir can't be None.")
+      benchmark_case_dir = benchmark_case.benchmark_case_dir
+
+      flag_file_path = benchmark_case_dir / BENCHMARK_FLAGFILE
+      with flag_file_path.open("r") as flag_file:
+        module_path = get_module_path(flag_file)
+
+      if module_path is None:
+        raise RuntimeError(
+            f"Can't find the module file in the flagfile: {flag_file_path}")
+      compilation_info = CompilationInfo(
+          model_name=benchmark_case.model_name,
+          model_tags=tuple(benchmark_case.model_tags),
+          model_source=category,
+          target_arch=benchmark_case.target_arch,
+          compile_tags=tuple(benchmark_case.bench_mode))
+      module_map[compilation_info] = (benchmark_case_dir /
+                                      module_path).resolve()
+
+  return module_map
+
+
+def legacy_get_module_map_and_build_log(args: argparse.Namespace):
+  module_map = get_module_map_from_benchmark_suite(
+      args.build_dir / benchmark_config.BENCHMARK_SUITE_REL_PATH)
+  return module_map, args.build_dir / NINJA_BUILD_LOG
+
+
+def alpha_get_module_map_and_build_log(args: argparse.Namespace):
+  module_map = get_module_map_from_generation_config(
+      serialized_gen_config=args.generation_config.open("r"),
+      e2e_test_artifacts_dir=args.e2e_test_artifacts_dir)
+  return module_map, args.build_log
+
+
 def parse_arguments():
   """Returns an argument parser with common options."""
 
-  def check_dir_path(path):
-    path = pathlib.Path(path)
+  def check_dir_path(path_str: str) -> pathlib.Path:
+    path = pathlib.Path(path_str)
     if path.is_dir():
       return path
     else:
-      raise argparse.ArgumentTypeError(path)
+      raise argparse.ArgumentTypeError(f"{path} is not a directory.")
+
+  def check_file_path(path_str: str) -> pathlib.Path:
+    path = pathlib.Path(path_str)
+    if path.is_file():
+      return path
+    else:
+      raise argparse.ArgumentTypeError(f"{path} is not a file.")
```

Comment:
Nit: the else isn't really necessary here since the other branch terminates

```suggestion
    if path.is_file():
      return path
    raise argparse.ArgumentTypeError(f"{path} is not a file.")
```

or if you think a guard statement is clearer

```suggestion
    if not path.is_file():
      raise argparse.ArgumentTypeError(f"{path} is not a file.")
    return path      
```

I think I actually like option two better